src/realm/object-store/sync/app_utils.cpp

```diff
@@ -22,18 +22,38 @@
 
 #include <external/json/json.hpp>
 
+#include <algorithm>
+
 namespace realm {
 namespace app {
 
+static bool string_iequals(const std::string& a, const std::string& b)
+{
+    return std::equal(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+        return tolower(a) == tolower(b);
+    });
+}
+
+util::Optional<std::pair<const std::string, std::string>>
+AppUtils::find_header(const std::string& key_name, const std::map<std::string, std::string>& search_map)
+{
+    for (auto current : search_map) {
```

Comment:
```suggestion
    for (auto&& current : search_map) {
```

Always, always, always use `for (auto&& name : iterable)`. There is basically never a reason to do anything else.